File: apps/ios/CrescendAI/Models/Student.swift
import Foundation
import SwiftData

@Model
final class Student {
    @Attribute(.unique) var appleUserId: String
    var email: String?
    var inferredLevel: String?

    // 6-dimension baselines (exponential moving average)
    var baselineDynamics: Double?
    var baselineTiming: Double?
    var baselinePedaling: Double?
    var baselineArticulation: Double?
    var baselinePhrasing: Double?
    var baselineInterpretation: Double?
    var baselineSessionCount: Int

    var explicitGoals: String?
    var lastSyncedAt: Date?
    var createdAt: Date

    @Relationship(deleteRule: .cascade, inverse: \PracticeSessionRecord.student)
    var sessions: [PracticeSessionRecord]

    init(
        appleUserId: String,
        email: String? = nil,
        inferredLevel: String? = nil,
        baselineSessionCount: Int = 0,
        explicitGoals: String? = nil
    ) {
        self.appleUserId = appleUserId
        self.email = email
        self.inferredLevel = inferredLevel
        self.baselineDynamics = nil
        self.baselineTiming = nil
        self.baselinePedaling = nil
        self.baselineArticulation = nil
        self.baselinePhrasing = nil
        self.baselineInterpretation = nil
        self.baselineSessionCount = baselineSessionCount
        self.explicitGoals = explicitGoals
        self.lastSyncedAt = nil
        self.createdAt = Date()
        self.sessions = []
    }
}

File: apps/api/src/harness/entities/student.ts
import { z } from "zod";

export const StudentSchema = z.object({
	studentId: z.string().min(1),
	inferredLevel: z.string().nullable().optional(),
	baselineDynamics: z.number().nullable().optional(),
	baselineTiming: z.number().nullable().optional(),
	baselinePedaling: z.number().nullable().optional(),
	baselineArticulation: z.number().nullable().optional(),
	baselinePhrasing: z.number().nullable().optional(),
	baselineInterpretation: z.number().nullable().optional(),
	baselineSessionCount: z.number().int().nonnegative(),
	explicitGoals: z.string().nullable().optional(),
	createdAt: z.string().datetime(),
	updatedAt: z.string().datetime(),
});

export type Student = z.infer<typeof StudentSchema>;

export function resolveStudent(input: { appleUserId: string }): {
	studentId: string;
} {
	return { studentId: input.appleUserId };
}
